### DISCUSSION TITLE
Error after running cabal build all

### DISCUSSION THREAD

**Discussion by jycappucino** [geniusyield/smart-order-router#4] [Q&A] [answered]
Hi guys,

I tried to build an SOR bot, but I got this error after running cabal build all:

Error: cabal: Could not resolve dependencies:
[__0] trying: atlas-cardano-0.3.0 (user goal)
[__1] next goal: cardano-ledger-core (dependency of atlas-cardano)
[__1] rejecting: cardano-ledger-core-1.2.0.0/installed-1Ob73qIWb7vK2Gwb7kzY3F
(does not contain library 'testlib', which is required by atlas-cardano)
[__1] fail (backjumping, conflict set: atlas-cardano, cardano-ledger-core)
After searching the rest of the dependency tree exhaustively, these were the
goals I've had most trouble fulfilling: atlas-cardano, cardano-ledger-core

**Comment by sourabhxyz** [comment, marked as answer]
Is this an issue within nix shell? If you are not using nix, please have your environment setup as per [this](https://github.com/geniusyield/atlas/blob/main/.github/workflows/haskell.yml) CI file (look for GHC, cabal versions and extra dependencies like libsodium, secp256k1).

I don't use nix and it's building fine for me.

**Reply by jycappucino** [reply]
Hello,

Sorry for the delay. Yes, I am in nix shell. To solve the problem, I installed atlas-cardano, i.e.,
`cabal install atlas-cardano --lib` and then `cabal build all`. 

The issue could potentially be resolved by installing the cardano-ledger-core - which I did not try since I resolved the issue as stated above. The current version of the cardano-ledger-core is 1.8.0.0, but the one used by the SOR is 1.2.0.0. The 1.2.0.0 is missing the `testlib`.
